pulsar-common/src/main/java/org/apache/pulsar/common/nar/NarUnpacker.java

```diff
@@ -69,7 +73,7 @@ static File doUnpackNar(final File nar, final File baseWorkingDirectory, Runnabl
             throws IOException {
         File parentDirectory = new File(baseWorkingDirectory, nar.getName() + "-unpacked");
         if (!parentDirectory.exists()) {
-            if (parentDirectory.mkdirs()) {
+            if (createDirWithSpecialPermission(parentDirectory, "rwxr-x---", true)) {
```

Comment:
I'd prefer to use constants for these perm settings. But it should not be a blocker.